plugin-flex-ts-template-v2/src/feature-library/contact-history/custom-components/ContactHistoryView.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { useState } from 'react';
+import { templates, Template } from '@twilio/flex-ui';
+import { useDispatch, useSelector } from 'react-redux';
+import { Button } from '@twilio-paste/core/button';
+import { Flex } from '@twilio-paste/core/flex';
+import { Heading } from '@twilio-paste/core/heading';
+import { Box } from '@twilio-paste/core/box';
+import { Table, THead, TBody, Tr, Th } from '@twilio-paste/core/table';
+import { AlertDialog } from '@twilio-paste/core/alert-dialog';
+import { DeleteIcon } from '@twilio-paste/icons/esm/DeleteIcon';
+
+import { clearContactList } from '../flex-hooks/state';
+import RecentContacts from '../utils/ContactsUtil';
+import ContactRecord from './ContactRecord';
+import { Contact } from '../types';
+import AppState from '../../../types/manager/AppState';
+import { reduxNamespace } from '../../../utils/state';
+import { StringTemplates } from '../flex-hooks/strings';
+
+const ContactHistory = () => {
+  const [confirmClearHistory, setConfirmClearHistory] = useState(false);
+
+  const contactData = useSelector((state: AppState) => {
+    return { contactList: state[reduxNamespace]?.contactHistory?.contactList };
+  });
+
+  const contactList = contactData?.contactList || [];
+  const dispatch = useDispatch();
+
+  const clearHistory = () => {
+    setConfirmClearHistory(false);
+    dispatch(clearContactList());
+    RecentContacts.clearContactList();
+  };
+
+  const closeClearHistory = () => setConfirmClearHistory(false);
+
+  return (
+    <Flex width="100%">
+      {!contactList || contactList.length === 0 ? (
+        <Flex vertical width="100%" hAlignContent="center" padding="space50">
+          <Heading as="h5" variant="heading50">
+            <Template source={templates[StringTemplates.NoHistory]} />
+          </Heading>
+        </Flex>
+      ) : (
+        <Flex vertical width="100%" grow shrink hAlignContent="center">
+          <Box overflowY="auto" width="100%" padding="space40">
+            <Table variant="borderless">
```

Comment:
I think the table should either have a border or go edge to edge like the schedule manager. Also try experimenting with alternating row colors, might make this easier to parse when long.

Reply:
Added border and striped